willow/image.py

```diff
@@ -147,42 +148,44 @@ def optimize(self, image_file, image_format: str):
 
         named_file_created = False
         try:
-            if isinstance(image_file, SpooledTemporaryFile):
-                file = image_file._file
+            if isinstance(image_file, (SpooledTemporaryFile, BytesIO)):
                 with NamedTemporaryFile(delete=False) as named_file:
-                    if hasattr(file, "getvalue"):  # e.g. BytesIO
-                        named_file.write(file.getvalue())
-                    else:  # e.g. BufferedRandom
-                        file.seek(0)
-                        named_file.write(file.read())
-                    file_path = named_file.name
-                named_file_created = True
-            elif isinstance(image_file, BytesIO):
-                with NamedTemporaryFile(delete=False) as named_file:
-                    named_file.write(image_file.getvalue())
+                    named_file_created = True
+
+                    image_file.seek(0)
+                    copyfileobj(image_file, named_file)
```

Comment:
👌🏼 
oh yes, this makes this whole block so much simpler

Reply:
`shutil.copyfileobj` is definitely one of those gems that makes things a lot cleaner. Sure, it's quite literally just buffered `.read` and `.write` calls - but it makes our code cleaner!